QMigrate/signal/local_mag/magnitude.py

```diff
@@ -0,0 +1,997 @@
+            Dist : bool
+                The (epicentral or hypocentral) distance between the station
+                and event.
+            Used : bool (== True)
+                Whether the observation meets all filter requirements.
+        all_mags : `pandas.DataFrame`
+            As for used_mags, but containing all observations from the input
+            magnitudes DataFrame, other than those which feature null values
+            for the signal or noise amplitude.
+
+        """
+
+        # Remove nan amplitude values
+        magnitudes.dropna(subset=[self.amp_feature, "Noise_amp"], inplace=True)
+        # magnitudes = magnitudes.loc[~np.isnan(amps), :]
+        # magnitudes = magnitudes.[~(pd.isnull(magnitudes[self.amp_feature]))]
+
+        # Apply noise filter.
+        if self.noise_filter:
+            amps = magnitudes[self.amp_feature].values
+            noise_amps = magnitudes["Noise_amp"].values
+            magnitudes["Noise_Filter"] = False
+            with np.errstate(invalid="ignore"):
+                magnitudes.loc[(amps > noise_amps * self.noise_filter),
+                               "Noise_Filter"] = True
+
+        # Apply trace filter
+        if self.trace_filter:
+            magnitudes["Trace_Filter"] = False
+            magnitudes.loc[magnitudes.index.str.contains(self.trace_filter),
+                           "Trace_Filter"] = True
+
+        # Apply station filter
+        if self.station_filter:
+            magnitudes["Station_Filter"] = True
+            for stn in list(self.station_filter):
+                magnitudes.loc[magnitudes.index.str.contains(f".{stn}.",
+                                                             regex=False),
+                               "Station_Filter"] = False
+
+        # Calculate distances
+        edist, zdist = magnitudes["epi_dist"], magnitudes["z_dist"]
+        if self.use_hyp_dist:
+            dist = np.sqrt(edist.values**2 + zdist.values**2)
+        else:
+            dist = edist.values
+
+        # Apply distance filter
+        if self.dist_filter:
+            magnitudes["Dist_Filter"] = False
+            magnitudes.loc[(dist <= self.dist_filter), "Dist_Filter"] = True
+
+        # Set distances; remove dist=0 values (logs do not like this)
+        dist[dist == 0.] = np.nan
+        magnitudes["Dist"] = dist
+
+        # Identify used mags (after applying all filters)
+        magnitudes["Used"] = True
+        if self.trace_filter:
+            magnitudes.loc[~magnitudes["Trace_Filter"], "Used"] = False
+        if self.station_filter:
+            magnitudes.loc[~magnitudes["Station_Filter"], "Used"] = False
+        if self.dist_filter:
+            magnitudes.loc[~magnitudes["Dist_Filter"], "Used"] = False
+        if self.pick_filter:
+            magnitudes.loc[~magnitudes["is_picked"], "Used"] = False
+        if self.noise_filter:
+            magnitudes.loc[~magnitudes["Noise_Filter"], "Used"] = False
```

Comment:
```suggestion
        if self.station_filter is not None:
            magnitudes.loc[~magnitudes["Station_Filter"], "Used"] = False
        if self.dist_filter:
            magnitudes.loc[~magnitudes["Dist_Filter"], "Used"] = False
        if self.pick_filter:
            magnitudes.loc[~magnitudes["is_picked"], "Used"] = False
        if self.noise_filter is not None:
            magnitudes.loc[~magnitudes["Noise_Filter"], "Used"] = False
```

Reply:
Ditto re. `self.trace_filter` being omitted, and checking `self.noise_filter` against `None`

Reply:
```suggestion
        if self.station_filter is not None:
            magnitudes.loc[~magnitudes["Station_Filter"], "Used"] = False
        if self.dist_filter:
            magnitudes.loc[~magnitudes["Dist_Filter"], "Used"] = False
        if self.pick_filter:
            magnitudes.loc[~magnitudes["is_picked"], "Used"] = False
        if self.noise_filter != 0.:
            magnitudes.loc[~magnitudes["Noise_Filter"], "Used"] = False
```